docs/specs/3-13-2023-commands-fit-n-finsh.md

```diff
@@ -0,0 +1,71 @@
+# 3/13/2023 - Commands Fit-and-Finish Proposal
+
+## Design
+
+Commands that interact with the core development workflow will be verb-based. This feels natural to the user as the user is building an application, provisioning infrastructure, and deploying their application to Azure -- which are all actions.
+
+For commands that are of configuration and management in-nature, it is fine to add a noun for a subcommand. For example, `azd env <action>` and `azd config <action>`.
+
+This means that the overall proposed design matches with what `azd` currently offers, and no changes need to occur.
+
+## `azd` usage output
+
+To help introduce user to the command layout, I propose slight changes to the current `azd` usage output (with UX input needed), that looks like the following:
+
+```
+Commands
+  Develop an app
+    init     	: Initialize a new application.
+    provision	: Provision the Azure resources for an application.
+    deploy   	: Deploy the application's code to Azure.
+    up       	: Initialize application, provision Azure resources, and deploy your project with a single command.
+
+  Monitor, test, and release your app
+    monitor  	: Monitor a deployed application.
+    pipeline 	: Manage and configure your deployment pipelines.
+
+  Authenticate with Azure
+    login    	: Log in to Azure.
```

Comment:
This does make the most common action a sub command as well though. azd auth login unless you would have azd auth with no arguments be to login?

Reply:
I agree with `auth`, because we need to expose `azd auth token` eventually.

How do we still feel about keeping under-the-covers aliasing? `azd login` will not be documented in the command layout, but `azd login` is an alias to `azd auth login`?

This is my personal preference. Auto-login of users is tricky, and we need to define the behavior for CI which I'm worried about.

Reply:
My current belief is that hidden aliasing is what we want. People that work with `az login` expect `azd login`, people that work with `gh` expect `gh auth login`.

Make `auth` the official approach. Make `azd login` a convenience alias. And I do mean alias. We don't do any extra support than forward to `azd auth login` (including helptext).

Reply:
Added to proposal.